### PR TITLE
EventHubs integration part 1

### PR DESCRIPTION
**Pull request by mssfang**
Follow up PR after "Eventhub-java: integrate event hubs into azure-sdk-for-java repo #3067"

[ X ] Checkstyle

Issue: Eventhubs: Integrate event hubs into azure-sdk-for-java repo #3065

### CODE REVIEW

File: .azure-pipelines/client.test.live.yml [hunk 1]
```diff
@@ -7,6 +7,8 @@ jobs:
           pomFilePath: './batch/data-plane/pom.xml'
```
Comment (conniey) [inline .azure-pipelines/client.test.live.yml]: Not sure if you meant to include this file in the PR. It doesn't seem related to checkstyle.

Comment (JonathanGiles) [reply .azure-pipelines/client.test.live.yml]: The better name for this PR would probably be 'EventHubs integration part 1' I would guess.

Comment (conniey) [reply .azure-pipelines/client.test.live.yml]: @mssfang Is it possible to split this PR into smaller ones? It'll make it easier to follow and CR. 190 files seems overwhelming, especially if some of them aren't to do with checkstyle.

Comment (JonathanGiles) [reply .azure-pipelines/client.test.live.yml]: I think for future reference this is a good idea - smaller PRs more frequently works better than one big PR less frequently. However, I've reviewed this PR now and nothing stands out as troublesome to me. Let's not worry about closing and recreating smaller PRs for this one - let's just get this in once there is another +1.

Comment (mssfang) [reply .azure-pipelines/client.test.live.yml]: Sorry for the big PR created. Will create multiple smaller PR in the future.

File: eventhubs/data-plane/azure-eventhubs/src/test/java/com/microsoft/azure/eventhubs/jproxy/ProxyNegotiationHandler.java [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+
+                attachment.isRead = false;
+                attachment.buffer.clear();
+                attachment.buffer.put(String.format("HTTP/1.1 502 %s\r\n\r\n", exc.getMessage()).getBytes());
+                attachment.buffer.flip();
+                clientSocket.write(attachment.buffer, attachment, ReadWriteHandler.this);
+            }
+        }
+    }
+}
```
Comment (conniey) [inline eventhubs/data-plane/azure-eventhubs/src/test/java/com/microsoft/azure/eventhubs/jproxy/ProxyNegotiationHandler.java]: nit: new-line at EOF.

Comment (conniey) [reply eventhubs/data-plane/azure-eventhubs/src/test/java/com/microsoft/azure/eventhubs/jproxy/ProxyNegotiationHandler.java]: and to the other proxy files you added below.

File: eventhubs/data-plane/pom.xml [hunk 3]
```diff
@@ -1,79 +1,66 @@
...
+      <id>azure-java-build-docs</id>
+      <url>${site.url}/site/${project.artifactId}</url>
+    </site>
+  </distributionManagement>
+
+  <scm>
+    <url>scm:git:https://github.com/Azure/azure-sdk-for-java</url>
+  </scm>
+
+  <properties>
```
Comment (conniey) [inline eventhubs/data-plane/pom.xml]: these values are already specified in the parent pom.client.xml?

Comment (mssfang) [reply eventhubs/data-plane/pom.xml]: Should I remove it? for all eventhubs' child modules? It is kind of conflict what I know from previous PR from Jonathan.
@JonathanGiles

Comment (JonathanGiles) [reply eventhubs/data-plane/pom.xml]: Connie's point is that these version numbers may be (or are) already specified in a parent pom, and therefore inherited here. You should try to not duplicate version properties in children as this can lead to inconsistent versions being used. You should check to ensure that they only exist in one place - the root parent pom file - and remove them from all other places.

File: eventhubs/data-plane/azure-eventhubs-eph/pom.xml [hunk 4]
```diff
@@ -1,40 +1,50 @@
...
-    </scm>
+  <dependencies>
+    <dependency>
+      <groupId>com.microsoft.azure</groupId>
+      <artifactId>azure-eventhubs</artifactId>
+      <version>${project.parent.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>com.microsoft.azure</groupId>
+      <artifactId>azure-storage</artifactId>
```
Comment (JamesBirdsall) [inline eventhubs/data-plane/azure-eventhubs-eph/pom.xml]: FYI: Event Processor Host has not been updated for the new Storage 10.x client yet. (We had one customer ask, then later close the issue stating that they had encountered reliability issues with 10.x, so we decided to wait a bit before doing the work for the API changes.)

File: eventhubs/data-plane/pom.xml (hunk 3 above)
Comment (mssfang) [reply eventhubs/data-plane/pom.xml]: ah, do you mean <version>1.0.0-SNAPSHOT</version> ?

Comment (JonathanGiles) [reply eventhubs/data-plane/pom.xml]: No - the content of the properties section that Connie has left her comment against.

Comment (mssfang) [reply eventhubs/data-plane/pom.xml]: remove line 11:   <version>2.0.0</version>, right?

Comment (mssfang) [reply eventhubs/data-plane/pom.xml]: i see, to delete the entire <properties>...</properties>

File: eventhubs/data-plane/azure-eventhubs-eph/pom.xml (hunk 4 above)
Comment (mssfang) [reply eventhubs/data-plane/azure-eventhubs-eph/pom.xml]: Thank you for sharing the information.

**Comment by mssfang** [comment]
If anyone has no another question to this PR. I will squash and merge it today. @JamesBirdsall  @JonathanGiles  @conniey

File: eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml [hunk 5]
```diff
@@ -27,17 +27,33 @@
   <!-- Public API already released with incorrect upper-case naming pattern -->
   <suppress checks="MethodName" files="AsymmetricEncryptionAlgorithm.java"/>
   <suppress checks="MethodName" files="KeyWrapAlgorithm.java"/>
+  <suppress checks="MethodName" files="PartitionSenderImpl.java"/>
```
Comment (conniey) [inline eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml]: This class is package-private. So it could be fixed.

File: eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml [hunk 6]
```diff
@@ -27,17 +27,33 @@
...
   <suppress checks="MethodName" files="SignatureEncoding.java"/>
   <suppress checks="MethodName" files="SymmetricEncryptionAlgorithm.java"/>
   <suppress checks="MethodName" files="RsaSignature.java"/>
 
   <!-- Public API already released with incorrect constant variable naming -->
   <suppress checks="ConstantName" files="AlgorithmResolver.java"/>
+  <suppress checks="ConstantName" files="AmqpErrorCode.java"/>
+  <suppress checks="ConstantName" files="BatchErrorCodeStrings.java"/>
+  <suppress checks="ConstantName" files="ConnectionStringBuilder.java"/>
```
Comment (conniey) [inline eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml]: Looking over your PR #3105 , the checkstyle-suppression comments I had may be similar. I made the same comment about ConnectionStringBuilder.java being able to be removed.

Since you're merging 3105 after this, I'd suggest removing this line instead of fixing it.

File: eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml [hunk 7]
```diff
@@ -27,17 +27,33 @@
...
+  <suppress checks="ConstantName" files="EventProcessorHost.java"/>
```
Comment (conniey) [inline eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml]: Remove this suppression for EventHostProcessor because it is fixed in #3105 .

File: eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml (hunk 5 above)
Comment (conniey) [reply eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml]: I believe this is fixed in #3105. So remove this line.

File: eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml (hunk 6 above)
Comment (mssfang) [reply eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml]: I updated checkstyle supression file. but not changes to the corresponding errors.
